scheduler/src/cook/mesos/share.clj

```diff
@@ -44,30 +44,59 @@
                    [[:db.fn/retractEntity resource]])
       (log/warn "Resource" type "for user" user "does not exist, could not retract"))))
 
-(defn- get-share-by-type
-  [db type user]
-  (let [type (resource-type->datomic-resource-type type)
-        query '[:find ?a
-                :in $ ?u ?t
-                :where
-                [?e :share/user ?u]
-                [?e :share/resource ?r]
-                [?r :resource/type ?t]
-                [?r :resource/amount ?a]]]
-    (or (-> (q query db user type) ffirst)
-        (-> (q query db default-user type) ffirst)
-        (Double/MAX_VALUE))))
+(let [query '[:find ?a
```

Comment:
nitpick: If you change the query to be `:find ?a .` it will return a single element so you don't need to do ffirst down belowhttp://docs.datomic.com/query.html#sec-5-8